core/src/main/kotlin/entity/channel/Channel.kt

```diff
@@ -55,7 +56,9 @@ public interface Channel : ChannelBehavior {
             GuildNews -> NewsChannel(data, kord)
             GuildForum -> ForumChannel(data, kord)
             PublicNewsThread -> NewsChannelThread(data, kord)
-            PrivateThread, PublicGuildThread -> TextChannelThread(data, kord)
+            PrivateThread, PublicGuildThread -> {
```

Comment:
Until discussion https://github.com/discord/discord-api-docs/discussions/5658 is answered or an efficient way to distinguish a forum thread from a text thread is found, this PR cannot be merge

Reply:
You can tell by checking if there is an applied tags as non-optional